src/neuroconv/datainterfaces/ecephys/neuralynx/neuralynxdatainterface.py

```diff
@@ -105,25 +39,73 @@ def initialize_in_spikeextractors(self, folder_path, verbose):
+        # add extractors for additional segments
+        for seg_idx in range(1, max_seg):
+            for i, filename in enumerate(self.ncs_files):
+                n_seg = recordings[i].neo_reader.segment_count(block_index=0)
+                if seg_idx <= n_seg:
+                    recordings.append(NeuralynxRecordingExtractor(filename=filename, seg_index=seg_idx))
+                else:
+                    # Use None as placeholder if the segment does not exist for requested ncs file
+                    recordings.append(None)
+
+        self.neo_readers = [r.neo_reader for r in recordings]
+        self.recording_extractor = self.Extractor(recordings)
+
+        gains = [recording.get_channel_gains()[0] for recording in recordings]
+        for recording in recordings:
+            recording.clear_channel_gains()
         self.recording_extractor.set_channel_gains(gains=gains)
 
     def add_recording_extractor_properties(self):
+        if hasattr(self.recording_extractor, "neo_reader"):
+            filtering = get_filtering_multi_channel(self.recording_extractor.neo_reader)
+        elif hasattr(self.recording_extractor, "neo_readers"):
+            filtering = get_filtering_single_channels(self.recording_extractor.neo_readers)
+        else:
+            raise ValueError("Unable to identify neo readers")
 
         try:
-            filtering = [get_filtering(filename) for filename in self.nsc_files]
             self.recording_extractor.set_property(key="filtering", values=filtering)
         except Exception:
             warnings.warn("filtering could not be extracted.")
 
     def get_metadata(self):
-        new_metadata = dict(NWBFile=get_metadata(self.source_data["folder_path"]))
-        return dict_deep_update(super().get_metadata(), new_metadata)
+        # extracting general session metadata exemplary from first recording
+        if hasattr(self.recording_extractor, "neo_reader"):
+            neo_metadata = extract_metadata_single_reader(self.recording_extractor.neo_reader)
+        elif hasattr(self.recording_extractor, "neo_readers"):
+            neo_metadata = extract_metadata_multi_reader(self.recording_extractor.neo_readers)
+        else:
+            raise ValueError("Unable to idenfity neo reader")
+
+        # remove filter related entries already covered by `add_recording_extractor_properties`
+        neo_metadata = {k: v for k, v in neo_metadata.items() if not k.lower().startswith("dsp")}
+
+        # map Neuralynx metadata to NWB
+        nwb_metadata = {"NWBFile": {}, "Ecephys": {"Device": []}}
+        if "session_id" in neo_metadata:
+            session_id = neo_metadata.pop("session_id", None)
+            if session_id is not None:
+                nwb_metadata["NWBFile"]["session_id"] = neo_metadata.pop("session_id")
+        if "SessionUUID" in neo_metadata:
+            nwb_metadata["NWBFile"]["identifier"] = neo_metadata.pop("SessionUUID")
```

Comment:
SessionUUID should map to session_id. "identifer" is generated from uuid4() by default and we can use that.

Reply:
I reverted back to storing the Neuralynx UUID as session_id